2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala

```diff
@@ -157,6 +158,26 @@ class CollectorService(
       case None => HttpResponse(404, entity = "404 not found")
     }
 
+
+  override def rootResponse: HttpResponse =
+    rootResponse(config.rootResponse)
+
+  def rootResponse(unknownRoute: Option[RootResponseConfig]): HttpResponse =
+    unknownRoute match {
+      case Some(c) =>
+        val statusCode = c.statusCode.getOrElse(404)
+        HttpResponse(
+          statusCode,
+          c.headers.map((headersMap: Map[String, String]) =>
+            headersMap
+              .map((tuple: (String, String)) => RawHeader(tuple._1, tuple._2))
+              .to[collection.immutable.Seq]
```

Comment:
you should be able to do:

```scala
headersMap.map { case (k, v) => RawHeader(k, v) }.toSeq
```

Reply:
Thanks for the pointer. I think I can switch to the case statement. The `toSeq` seems to have troubles due to its return type. It returns `scala.collection.Seq`, but `HttpResponse` is expecting a more specific type, `scala.collection.immutable.Seq`.